demos/common/utilities/sample_adu_jws_mbedtls.c

```diff
@@ -0,0 +1,796 @@
+            *pxBase64EncodedNSpan = pxPayload->_internal.xCoreReader.token.slice;
+
+            xResult = AzureIoTJSONReader_NextToken( pxPayload );
+        }
+        else if( AzureIoTJSONReader_TokenIsTextEqual( pxPayload, jws_e_json_value, sizeof( jws_e_json_value ) - 1 ) )
+        {
+            azureiotresultRETURN_IF_FAILED( AzureIoTJSONReader_NextToken( pxPayload ) );
+            *pxBase64EncodedESpan = pxPayload->_internal.xCoreReader.token.slice;
+
+            xResult = AzureIoTJSONReader_NextToken( pxPayload );
+        }
+        else if( AzureIoTJSONReader_TokenIsTextEqual( pxPayload, jws_alg_json_value, sizeof( jws_alg_json_value ) - 1 ) )
+        {
+            azureiotresultRETURN_IF_FAILED( AzureIoTJSONReader_NextToken( pxPayload ) );
+            *pxAlgSpan = pxPayload->_internal.xCoreReader.token.slice;
+
+            xResult = AzureIoTJSONReader_NextToken( pxPayload );
+        }
+        else
+        {
+            azureiotresultRETURN_IF_FAILED( AzureIoTJSONReader_NextToken( pxPayload ) );
+            azureiotresultRETURN_IF_FAILED( AzureIoTJSONReader_SkipChildren( pxPayload ) );
+            xResult = AzureIoTJSONReader_NextToken( pxPayload );
+        }
+    }
+
+    if( ( xResult != eAzureIoTSuccess ) && ( xResult != eAzureIoTErrorJSONReaderDone ) )
+    {
+        LogError( ( "[JWS] Parse Signing Key Payload Error: %i", xResult ) );
+        return xResult;
+    }
+
+    return 0;
+}
+
+static int32_t prvFindSHA( AzureIoTJSONReader_t * pxPayload,
+                           az_span * pxSHA )
+{
+    AzureIoTResult_t xResult = eAzureIoTSuccess;
+    AzureIoTJSONTokenType_t xJSONTokenType;
+
+    /*Begin object */
+    azureiotresultRETURN_IF_FAILED( AzureIoTJSONReader_NextToken( pxPayload ) );
+    /*Property Name */
+    azureiotresultRETURN_IF_FAILED( AzureIoTJSONReader_NextToken( pxPayload ) );
+
+    while( xResult == eAzureIoTSuccess )
+    {
+        if( AzureIoTJSONReader_TokenIsTextEqual( pxPayload, jws_sha256_json_value, sizeof( jws_sha256_json_value ) - 1 ) )
+        {
+            azureiotresultRETURN_IF_FAILED( AzureIoTJSONReader_NextToken( pxPayload ) );
+            break;
+        }
+        else
+        {
+            azureiotresultRETURN_IF_FAILED( AzureIoTJSONReader_NextToken( pxPayload ) );
+            azureiotresultRETURN_IF_FAILED( AzureIoTJSONReader_SkipChildren( pxPayload ) );
+            xResult = AzureIoTJSONReader_NextToken( pxPayload );
+        }
+    }
+
+    azureiotresultRETURN_IF_FAILED( AzureIoTJSONReader_TokenType( pxPayload, &xJSONTokenType ) );
+
+    if( ( xResult != eAzureIoTSuccess ) && ( xJSONTokenType == eAzureIoTJSONTokenSTRING ) )
+    {
+        LogError( ( "[JWS] Parse JSK JSON Payload Error: 0x%08x", xResult ) );
+        return xResult;
+    }
+
+    *pxSHA = pxPayload->_internal.xCoreReader.token.slice;
+
+    return 0;
+}
+
+uint32_t JWS_ManifestAuthenticate( const uint8_t * pucManifest,
```

Comment:
This is still really long - even if we have just simple helpers like as one example, `prvVerifyShaMatch` that could do from line 738 till end of function - that'd make this easier to follow.